build-aux/snap/snapcraft.yaml

```diff
@@ -28,172 +29,294 @@ license: GPL-3.0
+      if [ -d "${CRAFT_PROJECT_DIR}/build-aux" ]; then
+        cp "${CRAFT_PROJECT_DIR}"/build-aux/snap/local/apparmor/af_names.h .
       else
-        cp $SNAPCRAFT_PROJECT_DIR/snap/local/apparmor/af_names.h .
+        cp "${CRAFT_PROJECT_DIR}"/snap/local/apparmor/af_names.h .
       fi
-      make -j$(nproc)
-      mkdir -p $SNAPCRAFT_PART_INSTALL/usr/lib/snapd
-      cp -a apparmor_parser $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/
-      mkdir -p $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor
-      cp -a parser.conf $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor/
-      cd $SNAPCRAFT_PART_BUILD/profiles
-      make -j$(nproc)
-      mkdir -p $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor.d
-      cp -a apparmor.d/abi $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor.d/
-      cp -a apparmor.d/abstractions $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor.d/
-      cp -a apparmor.d/tunables $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor.d/
+      make -j"${CRAFT_PARALLEL_BUILD_COUNT}"
+      install -Dm755 -t "${CRAFT_PART_INSTALL}/usr/lib/snapd" apparmor_parser
+      install -Dm644 -t "${CRAFT_PART_INSTALL}/usr/lib/snapd/apparmor" parser.conf
+      cd "${CRAFT_PART_BUILD}/profiles"
+      make -j"${CRAFT_PARALLEL_BUILD_COUNT}"
+      mkdir -p "${CRAFT_PART_INSTALL}/usr/lib/snapd/apparmor.d"
+      cp -a apparmor.d/abi "${CRAFT_PART_INSTALL}/usr/lib/snapd/apparmor.d"
+      cp -a apparmor.d/abstractions "${CRAFT_PART_INSTALL}/usr/lib/snapd/apparmor.d"
+      cp -a apparmor.d/tunables "${CRAFT_PART_INSTALL}/usr/lib/snapd/apparmor.d"
+    prime:
+      # We are statically linked
+      - -usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/**
+      - -usr/include
+    <<:
+      - *dynamic-linker
+    override-prime: |
+      craftctl default
+      python3 "${CRAFT_PROJECT_DIR}/build-aux/snap/local/patch-dl.py" "/snap/snapd/current/usr/lib/${CRAFT_ARCH_TRIPLET_BUILD_FOR}/${DYNAMIC_LINKER}"
+
+  snapd:
+    plugin: nil
+    source: .
+    build-snaps:
+      - go/1.18/stable
+    after:
+      - apparmor
+    # these packages are needed to call mkversion.sh in override-pull, all other
+    # dependencies are installed using apt-get build-dep
+    build-packages:
+      - git
+      - dpkg-dev
+      - autoconf
+      - autoconf-archive
+      - automake
+      - xfslibs-dev
+      - libudev-dev
+      - libcap-dev
+      - libseccomp-dev
+      - pkg-config
+    override-pull: |
+      craftctl default
+      # set version, this needs dpkg-parsechangelog (from dpkg-dev) and git
+      craftctl set version="$(./mkversion.sh --output-only)"
+      ./get-deps.sh --skip-unused-check
+    override-build: |
+      ./mkversion.sh
+      cd "${CRAFT_PART_BUILD}/cmd"
+
+      autoreconf -fvi
+      ./configure --prefix=/usr --libexec=/usr/lib/snapd --without-unit-tests --enable-static-libapparmor --disable-host-binaries
+      make -j"${CRAFT_PARALLEL_BUILD_COUNT}"
+      make -j1 DESTDIR="${CRAFT_PART_INSTALL}" install
```

Comment:
Since it does not use libtool, it might be fine to run it parallel. Though not sure if there are any benefits. But I can remove the `-j1` at least. I added as it is common practice.

Reply:
Removed `-j1`. I did not add any `-j` because I am not sure it gives any benefit. But removing `-j1` makes it less confusing because there is no reason for it to be here.